core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -138,7 +138,10 @@ class DynamicConfigManager(private val zkClient: ZkClient,
             case Some(value: String) => value
             case _ => throw new IllegalArgumentException("Config change notification does not specify 'entity_name'. Received: " + json)
           }
-          configHandler(entityType).processConfigChanges(entity, AdminUtils.fetchEntityConfig(zkClient, entityType, entity))
+
+          val entityConfig = AdminUtils.fetchEntityConfig(zkClient, entityType, entity)
+          logger.info(s"Processing override for entityType: $entityType, entity: $entity with config: $entityConfig")
+          configHandler(entityType).processConfigChanges(entity, entityConfig)
```

Comment:
In DynamicConfigManager.startup(), it seems that we have a bug. Currently, it only reloads entities still in EntityConfigChangesPath. However, those entries will eventually be removed. So, on startup(), we should read from all existing entityPath directly and I don't see this logic.

BTW, this reloading logic is already covered in a generic ZkNodeChangeNotificationListener when we added the ACL support. We filed KAFKA-2547 to reuse it DynamicConfigManager. You can see if it's easier to do that as part of this jira. It may obviate some of the tests too.